src/app/code/community/Zendesk/Zendesk/Model/Observer.php

```diff
@@ -97,6 +97,20 @@ public function saveConfig(Varien_Event_Observer $observer)
                 Mage::getModel('core/config')->saveConfig('contacts/email/recipient_email', $oldEmail, $scope, $scopeId);
             }
         }
+
+        // If the zendesk domain is not found in the web widget snippet (wrapped with quotes), generate it again
+        $zDomain = Mage::getStoreConfig('zendesk/general/domain', $storeCode);
+        $widgetSnippet = Mage::getStoreConfig('zendesk/frontend_features/web_widget_code_snippet', $storeCode);
```

Comment:
The 2 lines above retrieve the Zendesk domain and Web Widget settings values from the sore config, in a similar way it was done a bit earlier in that method.